lib/pkgcloud/openstack/network/client/securityGroupRules.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * securityGroupRules.js: Instance methods for working with security group rules
+ * for Openstack Networking
+ *
+ * (C) 2015 Rackspace
+ *      Shaunak Kashyap
+ * MIT LICENSE
+ */
+
+var urlJoin = require('url-join');
+
+var securityGroupRulesResourcePath = '/security-group-rules';
+
+// Declaring variables for helper functions defined later
+var _convertSecurityGroupRuleToWireFormat;
+
+/**
+ * client.getSecurityGroupRules
+ *
+ * @description get the list of security group rules for an account
+ *
+ * @param {object|Function}   options
+ * @param {Function}          callback
+ */
+exports.getSecurityGroupRules = function (options, callback) {
+  var self = this;
+
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+
+  var getSecurityGroupRuleOpts = {
+    path: securityGroupRulesResourcePath
+  };
+
+  this._request(getSecurityGroupRuleOpts, function (err, body) {
+    if (err) {
+      return callback(err);
+    }
+    else if (!body || !body.security_group_rules || !(body.security_group_rules instanceof Array)) {
+      return new Error('Malformed API Response');
+    }
+
+    return callback(null, body.security_group_rules.map(function (securityGroupRule) {
+      return new self.models.SecurityGroupRule(self, securityGroupRule);
+    }));
+  });
+};
+
+/**
+ * client.getSecurityGroupRule
+ *
+ * @description get the details for a specific securityGroupRule
+ *
+ * @param {String|object}     securityGroupRule     the securityGroupRule or securityGroupRuleId
+ * @param callback
+ */
+exports.getSecurityGroupRule = function (securityGroupRule, callback) {
+  var securityGroupRuleId = securityGroupRule instanceof this.models.SecurityGroupRule ? securityGroupRule.id : securityGroupRule,
+    self = this;
+  self.emit('log::trace', 'Getting details for security group rule', securityGroupRuleId);
```

Comment:
Aside: we should think about better broad handling of logging. It appears sporadically in the code base.